### PR TITLE
[ALLUXIO-2391] Fix initial run of alluxio format fails because there is no logs dir

### PR DESCRIPTION
**Pull request by mtunique**
https://alluxio.atlassian.net/browse/ALLUXIO-2391

### CODE REVIEW

**Comment by mtunique** [comment]
alluxio-bot, check this please

**Comment by gpang** [comment]
ok to test

**Comment by gpang** [comment]
Jenkins, add to whitelist

**Comment by gpang** [comment]
Jenkins, test this please

File: bin/alluxio-workers.sh [hunk 1]
```diff
@@ -30,7 +30,9 @@ ALLUXIO_LIBEXEC_DIR=${ALLUXIO_LIBEXEC_DIR:-${DEFAULT_LIBEXEC_DIR}}
 . ${ALLUXIO_LIBEXEC_DIR}/alluxio-config.sh
 
 HOSTLIST=$(cat ${ALLUXIO_CONF_DIR}/workers | sed  "s/#.*$//;/^$/d")
-ALLUXIO_TASK_LOG="$(echo ${BIN} | sed 's/bin$//g')"logs/task.log
+ALLUXIO_TASK_LOG_DIR="${BIN//bin/}logs"
```
Comment (jsimsa) [inline bin/alluxio-workers.sh]: `ALLUXIO_TASK_LOG_DIR="${BIN}/../logs"` for consistency with how this is done elsewhere in this file

File: bin/alluxio-workers.sh [hunk 2]
```diff
@@ -30,7 +30,9 @@ ALLUXIO_LIBEXEC_DIR=${ALLUXIO_LIBEXEC_DIR:-${DEFAULT_LIBEXEC_DIR}}
...
+mkdir -p "$ALLUXIO_TASK_LOG_DIR"
```
Comment (jsimsa) [inline bin/alluxio-workers.sh]: `mkdir -p "${ALLUXIO_TASK_LOG_DIR}"

File: bin/alluxio-workers.sh [hunk 3]
```diff
@@ -30,7 +30,9 @@ ALLUXIO_LIBEXEC_DIR=${ALLUXIO_LIBEXEC_DIR:-${DEFAULT_LIBEXEC_DIR}}
...
+ALLUXIO_TASK_LOG="$ALLUXIO_TASK_LOG_DIR/task.log"
```
Comment (jsimsa) [inline bin/alluxio-workers.sh]: `ALLUXIO_TASK_LOG="${ALLUXIO_TASK_LOG_DIR}/task.log"`

File: bin/alluxio-workers.sh (hunk 1 above)
Comment (jsimsa) [inline bin/alluxio-workers.sh]: also, rename `ALLUXIO_TASK_LOG_DIR` to `ALLUXIO_LOG_DIR` since the directory contains all the logs, not just that task log

**Comment by mtunique** [comment]
@jsimsa Ok, I will change it.

**Comment by mtunique** [comment]
@jsimsa done

File: bin/alluxio-workers.sh [hunk 4]
```diff
@@ -30,7 +30,9 @@ ALLUXIO_LIBEXEC_DIR=${ALLUXIO_LIBEXEC_DIR:-${DEFAULT_LIBEXEC_DIR}}
 . ${ALLUXIO_LIBEXEC_DIR}/alluxio-config.sh
 
 HOSTLIST=$(cat ${ALLUXIO_CONF_DIR}/workers | sed  "s/#.*$//;/^$/d")
-ALLUXIO_TASK_LOG="$(echo ${BIN} | sed 's/bin$//g')"logs/task.log
+ALLUXIO_LOG_DIR="${BIN//bin/}logs"
```
Comment (jsimsa) [inline bin/alluxio-workers.sh]: `ALLUXIO_LOG_DIR="${BIN}/../logs"`

**Comment by mtunique** [comment]
@jsimsa done

File: bin/alluxio-workers.sh [hunk 5]
```diff
@@ -30,7 +30,9 @@ ALLUXIO_LIBEXEC_DIR=${ALLUXIO_LIBEXEC_DIR:-${DEFAULT_LIBEXEC_DIR}}
 . ${ALLUXIO_LIBEXEC_DIR}/alluxio-config.sh
 
 HOSTLIST=$(cat ${ALLUXIO_CONF_DIR}/workers | sed  "s/#.*$//;/^$/d")
-ALLUXIO_TASK_LOG="$(echo ${BIN} | sed 's/bin$//g')"logs/task.log
+ALLUXIO_LOG_DIR="${BIN}/../logs"
+mkdir -p "$ALLUXIO_LOG_DIR"
```
Comment (jsimsa) [inline bin/alluxio-workers.sh]: use ${ALLUXIO_LOG_DIR}

**Pull request merged**